### PR TITLE
[ENG-1429] [OSF Institutions] Shared SSO and The Policy Lab

### PR DESCRIPTION
**Pull request by cslzchen**
## Purpose

* Implement a feature that different institutions can share the same SSO
* Add the `thepolicylab` to `prod` using `brown`'s SSO of type `saml-shib`

## Changes

* Use a hard-coded map for identifying secondary institutions, which is an easy solution for now without re-design the institution model.
* As long as `brown` SSO succeeds, the authentication is successful. Errors with secondary institution only prevents the extra affiliation of the secondary one.
* Both institutions will be affiliated for `thepolicylab` users, which is a Product suggestion.
* Added API tests to cover both major and corner cases
* Integration tests with https://github.com/CenterForOpenScience/cas-overlay/pull/189 passed as well

### OSF

Run the script to add `thepolicylab` to `prod` institutions.

```
python3 -m scripts.populate_institutions -e prod -i thepolicylab
```

There is no need to update the `test` server.

### Shibboleth

Add the following line to the file `attribute-map.xml`.

```
<Attribute name="urn:oid:1.3.6.1.4.1.5923.1.5.1.1" id="isMemberOf"/>
```

### CAS

Add the following line to the `brown` block in the `institutions-auth.xsl ` file.

```
<isMemberOf><xsl:value-of select="//attribute[@name='isMemberOf']/@value"/></isMemberOf>
```

## QA Notes

N / A

## Documentation

N / A

## Side Effects

N / A

## Ticket

https://openscience.atlassian.net/browse/ENG-1429

### CODE REVIEW

File: api/institutions/authentication.py [hunk 1]
```diff
@@ -86,6 +123,46 @@ def authenticate(self, request):
...
+        # Check secondary institutions which uses the SSO of primary ones
+        secondary_institution = None
+        if provider['id'] in INSTITUTION_SHARED_SSO_MAP:
+            switch_map = INSTITUTION_SHARED_SSO_MAP[provider['id']]
+            criteria_type = switch_map.get('criteria')
+            if criteria_type == 'attribute':
+                attribute_name = switch_map.get('attribute')
+                attribute_value = provider['user'].get(attribute_name)
+                if attribute_value:
+                    secondary_institution_id = switch_map.get('institutions').get(attribute_value)
```
Comment (mfraezz) [inline api/institutions/authentication.py:135]: Minor: less important because it's a defined above rather than a variable param, but `{}.get('a').get('b')` results in an `AttributeError`, whereas `{}.get('a', {}).get('b')` does not.

Comment (cslzchen) [reply api/institutions/authentication.py:135]: Yep, I didn't do a check since it is defined, but `{}.get('a', {}).get('b')` is a very good suggestion. Let me update this.
